### ISSUE TITLE
error MSB4186: Invalid static method invocation syntax:

### ISSUE THREAD

**Issue by Bartmax** [dotnet/aspnetcore#31770]
### Describe the bug
When publishing I get this error:

```c#
error MSB4186: Invalid static method invocation syntax: "[MSBuild]::MakeRelative('$(MSBuildProjectDirectory)','$([MSBuild]::NormalizePath('wwwroot\/\videos\lkr_app-preview_1920x1080%2816,9%29_1.mp4'))')".
```

I can confirm the issue is because the filename is called: `lkr_app-preview_1920x1080(16,9)_1.mp4` and the issue is having a comma `,` on it's name.

### To Reproduce
I was able to reproduce by only including a file with a , on it's name: `FileWithComma,InTheName.txt`
Here's a repo with minimal repro commit: https://github.com/Bartmax/PublishBug/commit/d3a9b9631107baee64b54213c88bf290fbc813f8

### Exceptions (if any)
```c#
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21202.5\Sdks\Microsoft.NET.Sdk.Razor\build\netstandard2.0\Microsoft.NET.Sdk.Razor.StaticWebAssets.targets(540,9): error MSB4186: Invalid static method invocation syntax: "[MSBuild]::MakeRelative('$(MSBuildProjectDirectory)','$([MSBuild]::NormalizePath('wwwroot\/\FileWithComma,InTheName.txt'))')". Method '[MSBuild]::MakeRelative' not found. Static method invocation should be of the form: $([FullTypeName]::Method()), e.g. $([System.IO.Path]::Combine(`a`, `b`)). Check that all parameters are defined, are of the correct type, and are specified in the right order. [C:\Development\Experiments\PublishBug\Server\PublishBug.Server.csproj]
```

### Bonus
Visual studio new Git user experience is not able to add the file _alone_, must be done manually using git.

<details><summary><code>dotnet --info</code></summary>
### Further technical details
.NET SDK (reflecting any global.json):
 Version:   6.0.100-preview.3.21202.5
 Commit:    aee38a6dd4

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.100-preview.3.21202.5\

Host (useful for support):
  Version: 6.0.0-preview.3.21201.4
  Commit:  236cb21e3c

.NET SDKs installed:
  5.0.100-rc.1.20452.10 [C:\Program Files\dotnet\sdk]
  5.0.100-rc.2.20479.15 [C:\Program Files\dotnet\sdk]
  5.0.103 [C:\Program Files\dotnet\sdk]
  5.0.200-preview.20601.7 [C:\Program Files\dotnet\sdk]
  5.0.200-preview.21077.7 [C:\Program Files\dotnet\sdk]
  5.0.201 [C:\Program Files\dotnet\sdk]
  6.0.100-preview.3.21202.5 [C:\Program Files\dotnet\sdk]
</details>

_Error happens with NET 5 too._

**Comment by javiercn** [comment]
@Bartmax thanks for contacting us.

This might be a limitation regarded to this being done in MSBuild. I don't think there is an easy fix on our side (I might be wrong here) that doesn't involve writing a custom task.

If you want to try a fix for it in MSBuild, the right place would be [here](https://github.com/dotnet/sdk) since this is part of the Razor SDK. That said, please let us know if you are interested before you do any work so that we can help with some pointers here.

We are parking this in the backlog for now to gather more feedback and see if more people is affected by this issue.

**Comment by Bartmax** [comment]
Hey @javiercn that totally makes sense. I guess the issue is with [this task](https://github.com/dotnet/sdk/blob/c7d3f24be0c7d4fd28c734876bb16bc62bb13dfb/build_projects/dotnet-cli-build/MakeRelative.cs) 

I would totally love to fix this, I'm ready for those pointers.
Thank you!

**Comment by javiercn** [comment]
@Bartmax cool, let me think a bit about it and I'll update the issue next week.

I'm currently busy with other work, but I'll look at it once I have some time to spare.

**Comment by javiercn** [comment]
@Bartmax apologies for not getting back to you.

I believe this is fixed in preview7/rc1 in projects targeting 6.0, let me know otherwise.

(In short, we no longer have this code).
